funding/manager.go

```diff
@@ -2854,6 +3158,30 @@ func (f *Manager) handleFundingLocked(peer lnpeer.Peer,
 		return
 	}
 
+	// We'll need to store the received TLV alias for option_scid_alias
+	// channels. This will be used to provide route hints during invoice
+	// creation. In the zero-conf case it is also used to provide a
+	// ChannelUpdate to the remote peer. This is done before the call to
+	// InsertNextRevocation in case the call to PutAlias fails.
+	if channel.IsOptionScidAlias() || channel.ChanType.IsZeroConf() {
+		// If the AliasScid field is nil, we must fail out. We will
+		// most likely not be able to route through the peer.
+		if msg.AliasScid == nil {
+			log.Debugf("Consider closing ChannelID(%v), peer "+
+				"does not implement option-scid-alias "+
+				"properly", chanID)
+			return
+		}
+
+		// We'll store the AliasScid so that invoice creation can use
```

Comment:
I thought it was the other way around: we tell _them_ what scid alias we're using so they know how to forward payments that use that in the onion to us?

Reply:
So in this case, we only care so we can properly forward HTLCs to them.

Reply:
We tell them what scid alias to use in the invoice hop hints, and when we receive that alias in the onion, we'll forward to them. So here we just store the last received alias to use in our invoice hop hints